common/app/agents/DeeplyReadAgent.scala

```diff
@@ -1,18 +1,126 @@
 package agents
 
-import common.{Box, GuLogging}
-import model.dotcomrendering.{OnwardCollectionResponse, Trail}
-
-class DeeplyReadAgent extends GuLogging {
-  private val trailsBox = Box[Seq[Trail]](Seq())
-  def onwardsJourneyResponse = {
-    OnwardCollectionResponse(
-      heading = "Deeply read",
-      trails = trailsBox.get(),
+import com.gu.contentapi.client.model.v1.Content
+import com.gu.contentapi.client.utils.CapiModelEnrichment.RenderingFormat
+import common._
+import contentapi.ContentApiClient
+import model.ContentFormat
+import model.dotcomrendering.OnwardCollectionResponse
+import play.api.libs.json._
+import services.{OphanApi, OphanDeeplyReadItem}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+import model.dotcomrendering.Trail
+
+import java.lang.Throwable
+
+class DeeplyReadAgent(contentApiClient: ContentApiClient, ophanApi: OphanApi) extends GuLogging {
+
+  /*
+      We use a mutable map instead of a com.gu.Box, as the latter is a minimal wrapper.
```

Comment:
Is this comment true? Because I can see below that `deeplyReadItems` is of type `Box[Map[Edition, Seq[Trail]]]`

Reply:
Indeed, I think this comment dates from the original implementation and is no longer relevant.

Reply:
great catch! outdated documentation